engine/runtime-language-arrow/src/main/java/org/enso/interpreter/arrow/runtime/ArrowFixedSizeArrayBuilder.java

```diff
@@ -0,0 +1,353 @@
+package org.enso.interpreter.arrow.runtime;
+
+import com.oracle.truffle.api.CompilerDirectives;
+import com.oracle.truffle.api.dsl.Cached;
+import com.oracle.truffle.api.dsl.ImportStatic;
+import com.oracle.truffle.api.dsl.Specialization;
+import com.oracle.truffle.api.interop.*;
+import com.oracle.truffle.api.library.CachedLibrary;
+import com.oracle.truffle.api.library.ExportLibrary;
+import com.oracle.truffle.api.library.ExportMessage;
+import java.time.*;
+import org.enso.interpreter.arrow.LogicalLayout;
+
+@ExportLibrary(InteropLibrary.class)
+public final class ArrowFixedSizeArrayBuilder implements TruffleObject {
+  private final ByteBufferDirect buffer;
+  private final LogicalLayout unit;
+  private final int size;
+  private int index;
+
+  private boolean sealed;
+
+  public ArrowFixedSizeArrayBuilder(int size, LogicalLayout unit) {
+    this.size = size;
+    this.unit = unit;
+    this.buffer = ByteBufferDirect.forSize(size, unit);
+    this.index = 0;
+    this.sealed = false;
+  }
+
+  public LogicalLayout getUnit() {
+    return unit;
+  }
+
+  @ExportMessage
+  public boolean hasArrayElements() {
+    return true;
+  }
+
+  @ExportMessage
+  public boolean hasMembers() {
+    return true;
+  }
+
+  @ExportMessage
+  public boolean isMemberInvocable(String member) {
+    return switch (member) {
+      case "build", "append" -> !this.sealed;
+      default -> false;
+    };
+  }
+
+  @ExportMessage
+  Object getMembers(boolean includeInternal) throws UnsupportedMessageException {
+    return this;
+  }
+
+  @ExportMessage
+  @ImportStatic(LogicalLayout.class)
+  static class InvokeMember {
+    @Specialization(guards = "isDateLayout(receiver.getUnit())")
+    public static Object doInvokeDate(
```

Comment:
`doInvokeDate` seems like an left over.... but it is not! This means that the messages creates `ArrayFixedArrayDate` at the end.